CITATION.cff

```diff
@@ -0,0 +1,9 @@
+cff-version: 1.2.0
+message: If you use this guide, please cite it using these metadata.
+title: Rustc Dev Guide
```

Comment:
I think the official name is

```suggestion
title: Guide to Rustc Development
```

Reply:
Uh, hmm, but we've rarely used it, see https://github.com/search?q=org%3Arust-lang+%22Guide+to+Rustc+Development%22&type=code vs. https://github.com/search?q=org%3Arust-lang+%22rustc+dev+guide%22&type=code.
How about renaming it in this opportunity?

Reply:
Am happy with the rename, but would spell it...
> Rust Compiler Developer Guide

Reply:
Updated to "Rust Compiler Development Guide (rustc-dev-guide)".